### PR TITLE
Fix V2 parameter deserializer to recognize "

### PR DESCRIPTION
**Pull request by PerthCharernZero**
- Basic unit tests for 5 types of possible parameters in V2.
- V2 doc parameter in can be "formData", not just "form".

### CODE REVIEW

**Comment by PerthCharernZero** [comment]
Another reason for this PR:

I am submitting this from a different GitHub account to ensure that the repo settings works correctly for other contributors.

File: test/Microsoft.OpenApi.Readers.Tests/V2Tests/OpenApiParameterTests.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                // Form data parameter is currently not translated via LoadParameter.
+                // This design may be revisited and this unit test may likely change.
+                parameter.ShouldBeEquivalentTo(null);
+            }
+        }
+
+        [Fact]
+        public void ParseHeaderParameterShouldSucceed()
+        {
+            using (var stream = File.OpenRead(Path.Combine(SampleFolderPath, "headerParameter.yaml")))
```
Comment (xuzhg) [inline test/Microsoft.OpenApi.Readers.Tests/V2Tests/OpenApiParameterTests.cs]: // Arrange

File: test/Microsoft.OpenApi.Readers.Tests/V2Tests/OpenApiParameterTests.cs [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+                    });
+            }
+        }
+
+        [Fact]
+        public void ParseQueryParameterShouldSucceed()
+        {
+            using (var stream = File.OpenRead(Path.Combine(SampleFolderPath, "queryParameter.yaml")))
+            {
+                var yamlStream = new YamlStream();
```
Comment (xuzhg) [inline test/Microsoft.OpenApi.Readers.Tests/V2Tests/OpenApiParameterTests.cs]: // Arrange

File: test/Microsoft.OpenApi.Readers.Tests/V2Tests/OpenApiParameterTests.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                    });
+            }
+        }
+
+        [Fact]
+        public void ParseFormDataParameterShouldSucceed()
+        {
+            using (var stream = File.OpenRead(Path.Combine(SampleFolderPath, "formDataParameter.yaml")))
+            {
+                var yamlStream = new YamlStream();
```
Comment (xuzhg) [inline test/Microsoft.OpenApi.Readers.Tests/V2Tests/OpenApiParameterTests.cs]: // Arrange

**Pull request closed without merging**